src/ResourceManager/Network/Commands.Network/NetworkWatcher/GetAzureNetworkWatcherReachabilityProvidersListCommand.cs

```diff
@@ -0,0 +1,116 @@
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// 
+
+// Warning: This code was generated by a tool.
+// 
+// Changes to this file may cause incorrect behavior and will be lost if the
+// code is regenerated.
+
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network.Automation
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmNetworkWatcherReachabilityProvidersList", DefaultParameterSetName = "SetByName"), OutputType(typeof(PSAvailableProvidersList))]
+    public class GetAzureRmNetworkWatcherAvailableProviders : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The network watcher resource.",
+            ParameterSetName = "SetByResource")]
+        public PSNetworkWatcher NetworkWatcher { get; set; }
+
+        [Alias("ResourceName", "Name")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of network watcher.",
+            ParameterSetName = "SetByName")]
+        [ValidateNotNullOrEmpty]
+        public string NetworkWatcherName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the network watcher resource group.",
+            ParameterSetName = "SetByName")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Id of network watcher resource.",
+            ParameterSetName = "SetByResourceId")]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Optional Azure regions to scope the query to.",
+            ValueFromPipelineByPropertyName = true)]
+        public List<string> Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of the country.")]
+        public string Country { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of the state.",
+            ValueFromPipelineByPropertyName = true)]
```

Comment:
@irrogozh please remove `ValueFromPipelineByPropertyName` from this parameter for consistency with other parameters.

Reply:
Done